[C,pcsd/public/js/nodes-ember.js,2121]     var pacemaker_nodes_online = data["pacemaker_online"];
[C,pcsd/public/js/nodes-ember.js,2122]     var pacemaker_nodes_standby = data["pacemaker_standby"];
[C,pcsd/public/js/nodes-ember.js,2123] 
[C,pcsd/public/js/nodes-ember.js,2124]     var resources_on_nodes = {};
[C,pcsd/public/js/nodes-ember.js,2125]     var lc_on_nodes = {};
[C,pcsd/public/js/nodes-ember.js,2126]     $.each(data['node_list'], function(index, node) {
[C,pcsd/public/js/nodes-ember.js,2127]       nodes.push(node.name);
[C,pcsd/public/js/nodes-ember.js,2128] 
[C,pcsd/public/js/nodes-ember.js,2129]       resources_on_nodes[node.name] = [];
[C,pcsd/public/js/nodes-ember.js,2130]       $.each(Pcs.resourcesContainer.get('resource_map'), function(resource_id, resource_obj) {
[C,pcsd/public/js/nodes-ember.js,2131]         var nodes_running_on = resource_obj.get('nodes_running_on');
@@ -2066,31 +2150,45 @@ Pcs.nodesController = Ember.ArrayController.createWithMixins({
[C,pcsd/public/js/nodes-ember.js,2150]     var nodes_checked = {};
[C,pcsd/public/js/nodes-ember.js,2151]     var cur_node_holder = "";
[C,pcsd/public/js/nodes-ember.js,2152]     if (self.cur_node)
[C,pcsd/public/js/nodes-ember.js,2153]       cur_node_holder = self.cur_node.name;
[C,pcsd/public/js/nodes-ember.js,2154]     $.each(self.content, function (key, value) {
[C,pcsd/public/js/nodes-ember.js,2155]       if (value.checked)
[C,pcsd/public/js/nodes-ember.js,2156]         nodes_checked[value.name] = true;
[C,pcsd/public/js/nodes-ember.js,2157]     });
[C,pcsd/public/js/nodes-ember.js,2158] 
[C,pcsd/public/js/nodes-ember.js,2159]     if (data["nodes_utilization"]) {
[C,pcsd/public/js/nodes-ember.js,2160]       self.set("utilization_support", true);
[C,pcsd/public/js/nodes-ember.js,2161]     } else {
[C,pcsd/public/js/nodes-ember.js,2162]       self.set("utilization_support", false);
[C,pcsd/public/js/nodes-ember.js,2163]     }
[C,pcsd/public/js/nodes-ember.js,2164] 
[A,pcsd/public/js/nodes-ember.js,2165]     var is_sbd_enabled = false;
[A,pcsd/public/js/nodes-ember.js,2166]     var is_sbd_running = false;
[A,pcsd/public/js/nodes-ember.js,2167]     var sbd = null;
[A,pcsd/public/js/nodes-ember.js,2168]     Pcs.set("sbd_config", null);
[C,pcsd/public/js/nodes-ember.js,2169]     $.each(data['node_list'], function(_, node_obj) {
[A,pcsd/public/js/nodes-ember.js,2170]       if (node_obj["services"] && node_obj["services"]["sbd"]) {
[A,pcsd/public/js/nodes-ember.js,2171]         sbd = node_obj["services"]["sbd"];
[A,pcsd/public/js/nodes-ember.js,2172]         is_sbd_enabled = (is_sbd_enabled || sbd.enabled);
[A,pcsd/public/js/nodes-ember.js,2173]         is_sbd_running = (is_sbd_running || sbd.running);
[A,pcsd/public/js/nodes-ember.js,2174]       }
[A,pcsd/public/js/nodes-ember.js,2175] 
[A,pcsd/public/js/nodes-ember.js,2176]       if (node_obj["sbd_config"]) {
[A,pcsd/public/js/nodes-ember.js,2177]         Pcs.set("sbd_config", node_obj["sbd_config"]);
[A,pcsd/public/js/nodes-ember.js,2178]       }
[A,pcsd/public/js/nodes-ember.js,2179] 
[C,pcsd/public/js/nodes-ember.js,2180]       var node_id = node_obj.name;
[C,pcsd/public/js/nodes-ember.js,2181]       if ($.inArray(node_id, corosync_nodes_online) > -1) {
[C,pcsd/public/js/nodes-ember.js,2182]         corosync_online = true;
[C,pcsd/public/js/nodes-ember.js,2183]       } else {
[C,pcsd/public/js/nodes-ember.js,2184]         corosync_online = false;
[C,pcsd/public/js/nodes-ember.js,2185]       }
[C,pcsd/public/js/nodes-ember.js,2186] 
[C,pcsd/public/js/nodes-ember.js,2187]       if ($.inArray(node_id, pacemaker_nodes_online) > -1) {
[C,pcsd/public/js/nodes-ember.js,2188]         pacemaker_online = true;
[C,pcsd/public/js/nodes-ember.js,2189]       } else {
[C,pcsd/public/js/nodes-ember.js,2190]         pacemaker_online = false;
[C,pcsd/public/js/nodes-ember.js,2191]       }
[C,pcsd/public/js/nodes-ember.js,2192] 
[C,pcsd/public/js/nodes-ember.js,2193]       if ($.inArray(node_id, pacemaker_nodes_standby) > -1) {
[C,pcsd/public/js/nodes-ember.js,2194]         pacemaker_standby = true;
@@ -2141,81 +2239,88 @@ Pcs.nodesController = Ember.ArrayController.createWithMixins({
[C,pcsd/public/js/nodes-ember.js,2239]           node.set("pacemaker_daemon", node_obj["pacemaker"]);
[C,pcsd/public/js/nodes-ember.js,2240]           node.set("pacemaker_enabled", node_obj["pacemaker_enabled"]);
[C,pcsd/public/js/nodes-ember.js,2241]           node.set("pcsd_enabled", node_obj["pcsd_enabled"]);
[C,pcsd/public/js/nodes-ember.js,2242]           node.set("corosync", corosync_online);
[C,pcsd/public/js/nodes-ember.js,2243]           node.set("pacemaker", pacemaker_online);
[C,pcsd/public/js/nodes-ember.js,2244]           node.set("pacemaker_standby", pacemaker_standby);
[C,pcsd/public/js/nodes-ember.js,2245]           node.set("cur_node",false);
[C,pcsd/public/js/nodes-ember.js,2246]           node.set("running_resources", Pcs.getResourcesFromID($.unique(resources_on_nodes[node_id].sort().reverse())));
[C,pcsd/public/js/nodes-ember.js,2247]           node.set("location_constraints", lc_on_nodes[node_id].sort());
[C,pcsd/public/js/nodes-ember.js,2248]           node.set("uptime", node_obj["uptime"]);
[C,pcsd/public/js/nodes-ember.js,2249]           node.set("node_id", node_obj["id"]);
[C,pcsd/public/js/nodes-ember.js,2250]           node.set("node_attrs", node_attr);
[C,pcsd/public/js/nodes-ember.js,2251]           node.set("fence_levels", data["fence_levels"]);
[C,pcsd/public/js/nodes-ember.js,2252]           node.set("status", node_obj["status"]);
[C,pcsd/public/js/nodes-ember.js,2253]           node.set("utilization", utilization);
[A,pcsd/public/js/nodes-ember.js,2254]           node.set("services", node_obj["services"]);
[A,pcsd/public/js/nodes-ember.js,2255]           node.set("sbd_config", node_obj["sbd_config"]);
[C,pcsd/public/js/nodes-ember.js,2256]         }
[C,pcsd/public/js/nodes-ember.js,2257]       });
[C,pcsd/public/js/nodes-ember.js,2258] 
[C,pcsd/public/js/nodes-ember.js,2259]       if (found == false) {
[C,pcsd/public/js/nodes-ember.js,2260]         var node = Pcs.Clusternode.create({
[C,pcsd/public/js/nodes-ember.js,2261]           name: node_id,
[C,pcsd/public/js/nodes-ember.js,2262]           authorized:  authorized,
[C,pcsd/public/js/nodes-ember.js,2263]           up: up_status,
[C,pcsd/public/js/nodes-ember.js,2264]           pcsd: pcsd_daemon && authorized,
[C,pcsd/public/js/nodes-ember.js,2265]           corosync_daemon: node_obj["corosync"],
[C,pcsd/public/js/nodes-ember.js,2266]           corosync_enabled: node_obj["corosync_enabled"],
[C,pcsd/public/js/nodes-ember.js,2267]           pacemaker_daemon: node_obj["pacemaker"],
[C,pcsd/public/js/nodes-ember.js,2268]           pacemaker_enabled: node_obj["pacemaker_enabled"],
[C,pcsd/public/js/nodes-ember.js,2269]           pcsd_enabled: node_obj["pcsd_enabled"],
[C,pcsd/public/js/nodes-ember.js,2270]           corosync: corosync_online,
[C,pcsd/public/js/nodes-ember.js,2271]           pacemaker: pacemaker_online,
[C,pcsd/public/js/nodes-ember.js,2272]           pacemaker_standby: pacemaker_standby,
[C,pcsd/public/js/nodes-ember.js,2273]           cur_node: false,
[C,pcsd/public/js/nodes-ember.js,2274]           running_resources: Pcs.getResourcesFromID($.unique(resources_on_nodes[node_id].sort().reverse())),
[C,pcsd/public/js/nodes-ember.js,2275]           location_constraints: lc_on_nodes[node_id].sort(),
[C,pcsd/public/js/nodes-ember.js,2276]           uptime: node_obj["uptime"],
[C,pcsd/public/js/nodes-ember.js,2277]           node_id: node_obj["id"],
[C,pcsd/public/js/nodes-ember.js,2278]           node_attrs: node_attr,
[C,pcsd/public/js/nodes-ember.js,2279]           fence_levels: data["fence_levels"],
[C,pcsd/public/js/nodes-ember.js,2280]           status: node_obj["status"],
[D,pcsd/public/js/nodes-ember.js,2181]           utilization: utilization
[A,pcsd/public/js/nodes-ember.js,2281]           utilization: utilization,
[A,pcsd/public/js/nodes-ember.js,2282]           services: node_obj["services"],
[A,pcsd/public/js/nodes-ember.js,2283]           sbd_config: node_obj["sbd_config"]
[C,pcsd/public/js/nodes-ember.js,2284]         });
[C,pcsd/public/js/nodes-ember.js,2285]       }
[C,pcsd/public/js/nodes-ember.js,2286]       var pathname = window.location.pathname.split('/');
[C,pcsd/public/js/nodes-ember.js,2287] 
[C,pcsd/public/js/nodes-ember.js,2288]       if (cur_node_holder == "") {
[C,pcsd/public/js/nodes-ember.js,2289]         cur_node_name = Pcs.opening_node;
[C,pcsd/public/js/nodes-ember.js,2290]       } else {
[C,pcsd/public/js/nodes-ember.js,2291]         cur_node_name = cur_node_holder;
[C,pcsd/public/js/nodes-ember.js,2292]       }
[C,pcsd/public/js/nodes-ember.js,2293]       if (node.name == cur_node_name) {
[C,pcsd/public/js/nodes-ember.js,2294]         node.set("cur_node",true);
[C,pcsd/public/js/nodes-ember.js,2295]         self.set("cur_node", node);
[C,pcsd/public/js/nodes-ember.js,2296]       }
[C,pcsd/public/js/nodes-ember.js,2297] 
[C,pcsd/public/js/nodes-ember.js,2298]       if (nodes_checked[node.name])
[C,pcsd/public/js/nodes-ember.js,2299]         node.set("checked",true);
[C,pcsd/public/js/nodes-ember.js,2300] 
[C,pcsd/public/js/nodes-ember.js,2301]       if (found == false)
[C,pcsd/public/js/nodes-ember.js,2302]         self.pushObject(node);
[C,pcsd/public/js/nodes-ember.js,2303]     });
[C,pcsd/public/js/nodes-ember.js,2304]     if (self.content && self.content.length > 0 && self.cur_node == null) {
[C,pcsd/public/js/nodes-ember.js,2305]       self.set("cur_node", self.content[0]);
[C,pcsd/public/js/nodes-ember.js,2306]       self.content[0].set("cur_node",true);
[C,pcsd/public/js/nodes-ember.js,2307]     }
[C,pcsd/public/js/nodes-ember.js,2308] 
[A,pcsd/public/js/nodes-ember.js,2309]     Pcs.set("is_sbd_enabled", is_sbd_enabled);
[A,pcsd/public/js/nodes-ember.js,2310]     Pcs.set("is_sbd_running", is_sbd_running);
[A,pcsd/public/js/nodes-ember.js,2311] 
[C,pcsd/public/js/nodes-ember.js,2312]     nodesToRemove = [];
[C,pcsd/public/js/nodes-ember.js,2313]     $.each(self.content, function (key, node) {
[C,pcsd/public/js/nodes-ember.js,2314]       found = false;
[C,pcsd/public/js/nodes-ember.js,2315]       $.each(nodes, function (k,v) {
[C,pcsd/public/js/nodes-ember.js,2316]         if (v == node.name)
[C,pcsd/public/js/nodes-ember.js,2317]           found = true;
[C,pcsd/public/js/nodes-ember.js,2318]       });
[C,pcsd/public/js/nodes-ember.js,2319]       if (!found) {
[C,pcsd/public/js/nodes-ember.js,2320]         nodesToRemove.push(node);
[C,pcsd/public/js/nodes-ember.js,2321]       }
[C,pcsd/public/js/nodes-ember.js,2322]     });
[C,pcsd/public/js/nodes-ember.js,2323] 
[C,pcsd/public/js/nodes-ember.js,2324]     $.each(nodesToRemove, function(k,v) {
[C,pcsd/public/js/nodes-ember.js,2325]       self.content.removeObject(v);
[C,pcsd/public/js/nodes-ember.js,2326]     });
@@ -2710,18 +2710,200 @@ Ember.Handlebars.helper('selector-helper', function (content, value, place_holde
[C,pcsd/public/js/pcsd.js,2710]   var line;
[C,pcsd/public/js/pcsd.js,2711]   if (place_holder) {
[C,pcsd/public/js/pcsd.js,2712]     out += '<option value="">' + place_holder + '</option>'; 
[C,pcsd/public/js/pcsd.js,2713]   }
[C,pcsd/public/js/pcsd.js,2714]   $.each(content, function(_, opt){
[C,pcsd/public/js/pcsd.js,2715]     line = '<option value="' + opt["value"] + '"';
[C,pcsd/public/js/pcsd.js,2716]     if (value == opt["value"]) {
[C,pcsd/public/js/pcsd.js,2717]       line += ' selected="selected"'
[C,pcsd/public/js/pcsd.js,2718]     }
[C,pcsd/public/js/pcsd.js,2719]     line += ">" + Handlebars.Utils.escapeExpression(opt["name"]) + "</option>";
[C,pcsd/public/js/pcsd.js,2720]     out += line + "\n";
[C,pcsd/public/js/pcsd.js,2721]   });
[C,pcsd/public/js/pcsd.js,2722]   return new Handlebars.SafeString(out);
[C,pcsd/public/js/pcsd.js,2723] });
[C,pcsd/public/js/pcsd.js,2724] 
[A,pcsd/public/js/pcsd.js,2725] Ember.Handlebars.helper('bool-to-icon', function(value, options) {
[A,pcsd/public/js/pcsd.js,2726]   var out = '<span class="sprites inverted ';
[A,pcsd/public/js/pcsd.js,2727]   if (typeof(value) == 'undefined' || value == null) {
[A,pcsd/public/js/pcsd.js,2728]     out += "questionmarkdark";
[A,pcsd/public/js/pcsd.js,2729]   } else if (value) {
[A,pcsd/public/js/pcsd.js,2730]     out += "checkdark"
[A,pcsd/public/js/pcsd.js,2731]   } else {
[A,pcsd/public/js/pcsd.js,2732]     out += "Xdark"
[A,pcsd/public/js/pcsd.js,2733]   }
[A,pcsd/public/js/pcsd.js,2734]   return new Handlebars.SafeString(out + '">&nbsp;</span>');
[A,pcsd/public/js/pcsd.js,2735] });
[A,pcsd/public/js/pcsd.js,2736] 
[C,pcsd/public/js/pcsd.js,2737] function nl2br(text) {
[C,pcsd/public/js/pcsd.js,2738]   return text.replace(/(?:\r\n|\r|\n)/g, '<br />');
[C,pcsd/public/js/pcsd.js,2739] }
[A,pcsd/public/js/pcsd.js,2740] 
[A,pcsd/public/js/pcsd.js,2741] function enable_sbd(dialog) {
[A,pcsd/public/js/pcsd.js,2742]   ajax_wrapper({
[A,pcsd/public/js/pcsd.js,2743]     type: 'POST',
[A,pcsd/public/js/pcsd.js,2744]     url: get_cluster_remote_url() + "remote_enable_sbd",
[A,pcsd/public/js/pcsd.js,2745]     data: dialog.find("#enable_sbd_form").serialize(),
[A,pcsd/public/js/pcsd.js,2746]     timeout: pcs_timeout,
[A,pcsd/public/js/pcsd.js,2747]     success: function() {
[A,pcsd/public/js/pcsd.js,2748]       dialog.parent().find("#enable_sbd_btn").button(
[A,pcsd/public/js/pcsd.js,2749]         "option", "disabled", false
[A,pcsd/public/js/pcsd.js,2750]       );
[A,pcsd/public/js/pcsd.js,2751]       dialog.dialog("close");
[A,pcsd/public/js/pcsd.js,2752]       alert(
[A,pcsd/public/js/pcsd.js,2753]         'SBD enabled! You have to restart cluster in order to apply changes.'
[A,pcsd/public/js/pcsd.js,2754]       );
[A,pcsd/public/js/pcsd.js,2755]       Pcs.update();
[A,pcsd/public/js/pcsd.js,2756]     },
[A,pcsd/public/js/pcsd.js,2757]     error: function (xhr, status, error) {
[A,pcsd/public/js/pcsd.js,2758]       dialog.parent().find("#enable_sbd_btn").button(
[A,pcsd/public/js/pcsd.js,2759]         "option", "disabled", false
[A,pcsd/public/js/pcsd.js,2760]       );
[A,pcsd/public/js/pcsd.js,2761]       xhr.responseText = xhr.responseText.replace(
[A,pcsd/public/js/pcsd.js,2762]         "--skip-offline", "option 'ignore offline nodes'"
[A,pcsd/public/js/pcsd.js,2763]       );
[A,pcsd/public/js/pcsd.js,2764]       alert(
[A,pcsd/public/js/pcsd.js,2765]         ajax_simple_error(xhr, status, error)
[A,pcsd/public/js/pcsd.js,2766]       );
[A,pcsd/public/js/pcsd.js,2767]     }
[A,pcsd/public/js/pcsd.js,2768]   });
[A,pcsd/public/js/pcsd.js,2769] }
[A,pcsd/public/js/pcsd.js,2770] 
[A,pcsd/public/js/pcsd.js,2771] function enable_sbd_dialog(node_list) {
[A,pcsd/public/js/pcsd.js,2772]   var buttonsOpts = [
[A,pcsd/public/js/pcsd.js,2773]     {
[A,pcsd/public/js/pcsd.js,2774]       text: "Enable SBD",
[A,pcsd/public/js/pcsd.js,2775]       id: "enable_sbd_btn",
[A,pcsd/public/js/pcsd.js,2776]       click: function() {
[A,pcsd/public/js/pcsd.js,2777]         var dialog = $(this);
[A,pcsd/public/js/pcsd.js,2778]         dialog.parent().find("#enable_sbd_btn").button(
[A,pcsd/public/js/pcsd.js,2779]           "option", "disabled", true
[A,pcsd/public/js/pcsd.js,2780]         );
[A,pcsd/public/js/pcsd.js,2781]         enable_sbd(dialog);
[A,pcsd/public/js/pcsd.js,2782]       }
[A,pcsd/public/js/pcsd.js,2783]     },
[A,pcsd/public/js/pcsd.js,2784]     {
[A,pcsd/public/js/pcsd.js,2785]       text:"Cancel",
[A,pcsd/public/js/pcsd.js,2786]       click: function () {
[A,pcsd/public/js/pcsd.js,2787]         $(this).dialog("close");
[A,pcsd/public/js/pcsd.js,2788]       }
[A,pcsd/public/js/pcsd.js,2789]     }
[A,pcsd/public/js/pcsd.js,2790]   ];
[A,pcsd/public/js/pcsd.js,2791] 
[A,pcsd/public/js/pcsd.js,2792]   var dialog_obj = $("#enable_sbd_dialog").dialog({title: 'Enable SBD',
[A,pcsd/public/js/pcsd.js,2793]     modal: true, resizable: false,
[A,pcsd/public/js/pcsd.js,2794]     width: 'auto',
[A,pcsd/public/js/pcsd.js,2795]     buttons: buttonsOpts
[A,pcsd/public/js/pcsd.js,2796]   });
[A,pcsd/public/js/pcsd.js,2797] 
[A,pcsd/public/js/pcsd.js,2798]   dialog_obj.keypress(function(e) {
[A,pcsd/public/js/pcsd.js,2799]     if (
[A,pcsd/public/js/pcsd.js,2800]       e.keyCode == $.ui.keyCode.ENTER &&
[A,pcsd/public/js/pcsd.js,2801]       !dialog_obj.parent().find("#enable_sbd_btn").button("option", "disabled")
[A,pcsd/public/js/pcsd.js,2802]     ) {
[A,pcsd/public/js/pcsd.js,2803]       dialog_obj.parent().find("#enable_sbd_btn").trigger("click");
[A,pcsd/public/js/pcsd.js,2804]       return false;
[A,pcsd/public/js/pcsd.js,2805]     }
[A,pcsd/public/js/pcsd.js,2806]   });
[A,pcsd/public/js/pcsd.js,2807]   dialog_obj.find('#watchdog_table').empty();
[A,pcsd/public/js/pcsd.js,2808]   $.each(node_list, function(_, node) {
[A,pcsd/public/js/pcsd.js,2809]     dialog_obj.find("#watchdog_table").append(
[A,pcsd/public/js/pcsd.js,2810]       '<tr>' +
[A,pcsd/public/js/pcsd.js,2811]         '<td>' +
[A,pcsd/public/js/pcsd.js,2812]           node + ':' +
[A,pcsd/public/js/pcsd.js,2813]         '</td>' +
[A,pcsd/public/js/pcsd.js,2814]         '<td>' +
[A,pcsd/public/js/pcsd.js,2815]           '<input ' +
[A,pcsd/public/js/pcsd.js,2816]             'type="text" ' +
[A,pcsd/public/js/pcsd.js,2817]             'placeholder="/dev/watchdog" ' +
[A,pcsd/public/js/pcsd.js,2818]             'name="watchdog[' + node + ']" ' +
[A,pcsd/public/js/pcsd.js,2819]           '/>' +
[A,pcsd/public/js/pcsd.js,2820]         '</td>' +
[A,pcsd/public/js/pcsd.js,2821]       '</tr>'
[A,pcsd/public/js/pcsd.js,2822]     )
[A,pcsd/public/js/pcsd.js,2823]   });
[A,pcsd/public/js/pcsd.js,2824] }
[A,pcsd/public/js/pcsd.js,2825] 
[A,pcsd/public/js/pcsd.js,2826] function disable_sbd(dialog) {
[A,pcsd/public/js/pcsd.js,2827]   ajax_wrapper({
[A,pcsd/public/js/pcsd.js,2828]     type: 'POST',
[A,pcsd/public/js/pcsd.js,2829]     url: get_cluster_remote_url() + "remote_disable_sbd",
[A,pcsd/public/js/pcsd.js,2830]     data: dialog.find("#disable_sbd_form").serialize(),
[A,pcsd/public/js/pcsd.js,2831]     timeout: pcs_timeout,
[A,pcsd/public/js/pcsd.js,2832]     success: function() {
[A,pcsd/public/js/pcsd.js,2833]       dialog.parent().find("#disable_sbd_btn").button(
[A,pcsd/public/js/pcsd.js,2834]         "option", "disabled", false
[A,pcsd/public/js/pcsd.js,2835]       );
[A,pcsd/public/js/pcsd.js,2836]       dialog.dialog("close");
[A,pcsd/public/js/pcsd.js,2837]       alert(
[A,pcsd/public/js/pcsd.js,2838]         'SBD disabled! You have to restart cluster in order to apply changes.'
[A,pcsd/public/js/pcsd.js,2839]       );
[A,pcsd/public/js/pcsd.js,2840]       Pcs.update();
[A,pcsd/public/js/pcsd.js,2841]     },
[A,pcsd/public/js/pcsd.js,2842]     error: function (xhr, status, error) {
[A,pcsd/public/js/pcsd.js,2843]       dialog.parent().find("#disable_sbd_btn").button(
[A,pcsd/public/js/pcsd.js,2844]         "option", "disabled", false
[A,pcsd/public/js/pcsd.js,2845]       );
[A,pcsd/public/js/pcsd.js,2846]       xhr.responseText = xhr.responseText.replace(
[A,pcsd/public/js/pcsd.js,2847]         "--skip-offline", "option 'ignore offline nodes'"
[A,pcsd/public/js/pcsd.js,2848]       );
[A,pcsd/public/js/pcsd.js,2849]       alert(ajax_simple_error(xhr, status, error));
[A,pcsd/public/js/pcsd.js,2850]     }
[A,pcsd/public/js/pcsd.js,2851]   });
[A,pcsd/public/js/pcsd.js,2852] }
[A,pcsd/public/js/pcsd.js,2853] 
[A,pcsd/public/js/pcsd.js,2854] function disable_sbd_dialog() {
[A,pcsd/public/js/pcsd.js,2855]   var buttonsOpts = [
[A,pcsd/public/js/pcsd.js,2856]     {
[A,pcsd/public/js/pcsd.js,2857]       text: "Disable SBD",
[A,pcsd/public/js/pcsd.js,2858]       id: "disable_sbd_btn",
[A,pcsd/public/js/pcsd.js,2859]       click: function() {
[A,pcsd/public/js/pcsd.js,2860]         var dialog = $(this);
[A,pcsd/public/js/pcsd.js,2861]         dialog.parent().find("#disable_sbd_btn").button(
[A,pcsd/public/js/pcsd.js,2862]           "option", "disabled", true
[A,pcsd/public/js/pcsd.js,2863]         );
[A,pcsd/public/js/pcsd.js,2864]         disable_sbd(dialog);
[A,pcsd/public/js/pcsd.js,2865]       }
[A,pcsd/public/js/pcsd.js,2866]     },
[A,pcsd/public/js/pcsd.js,2867]     {
[A,pcsd/public/js/pcsd.js,2868]       text:"Cancel",
[A,pcsd/public/js/pcsd.js,2869]       click: function () {
[A,pcsd/public/js/pcsd.js,2870]         $(this).dialog("close");
[A,pcsd/public/js/pcsd.js,2871]       }
[A,pcsd/public/js/pcsd.js,2872]     }
[A,pcsd/public/js/pcsd.js,2873]   ];
[A,pcsd/public/js/pcsd.js,2874] 
[A,pcsd/public/js/pcsd.js,2875]   $("#disable_sbd_dialog").dialog({
[A,pcsd/public/js/pcsd.js,2876]     title: 'Disable SBD',
[A,pcsd/public/js/pcsd.js,2877]     modal: true, resizable: false,
[A,pcsd/public/js/pcsd.js,2878]     width: 'auto',
[A,pcsd/public/js/pcsd.js,2879]     buttons: buttonsOpts
[A,pcsd/public/js/pcsd.js,2880]   });
[A,pcsd/public/js/pcsd.js,2881] }
[A,pcsd/public/js/pcsd.js,2882] 
[A,pcsd/public/js/pcsd.js,2883] function sbd_status_dialog() {
[A,pcsd/public/js/pcsd.js,2884]   var buttonsOpts = [
[A,pcsd/public/js/pcsd.js,2885]     {
[A,pcsd/public/js/pcsd.js,2886]       text: "Enable SBD",
[A,pcsd/public/js/pcsd.js,2887]       click: function() {
[A,pcsd/public/js/pcsd.js,2888]         enable_sbd_dialog(Pcs.nodesController.get_node_name_list());
[A,pcsd/public/js/pcsd.js,2889]       }
[A,pcsd/public/js/pcsd.js,2890]     },
[A,pcsd/public/js/pcsd.js,2891]     {
[A,pcsd/public/js/pcsd.js,2892]       text: "Disable SBD",
[A,pcsd/public/js/pcsd.js,2893]       click: disable_sbd_dialog
[A,pcsd/public/js/pcsd.js,2894]     },
[A,pcsd/public/js/pcsd.js,2895]     {
[A,pcsd/public/js/pcsd.js,2896]       text:"Close",
[A,pcsd/public/js/pcsd.js,2897]       click: function () {
[A,pcsd/public/js/pcsd.js,2898]         $(this).dialog("close");
[A,pcsd/public/js/pcsd.js,2899]       }
[A,pcsd/public/js/pcsd.js,2900]     }
[A,pcsd/public/js/pcsd.js,2901]   ];
[A,pcsd/public/js/pcsd.js,2902] 
[A,pcsd/public/js/pcsd.js,2903]   $("#sbd_status_dialog").dialog({
[A,pcsd/public/js/pcsd.js,2904]     title: 'SBD',
[A,pcsd/public/js/pcsd.js,2905]     modal: true, resizable: false,
[A,pcsd/public/js/pcsd.js,2906]     width: 'auto',
[A,pcsd/public/js/pcsd.js,2907]     buttons: buttonsOpts
[A,pcsd/public/js/pcsd.js,2908]   });
[A,pcsd/public/js/pcsd.js,2909] }
@@ -57,31 +57,33 @@ def remote(params, request, auth_user)
[C,pcsd/remote.rb,57]       :remove_nodes => method(:remote_remove_nodes),
[C,pcsd/remote.rb,58]       :remove_node => method(:remote_remove_node),
[C,pcsd/remote.rb,59]       :cluster_destroy => method(:cluster_destroy),
[C,pcsd/remote.rb,60]       :get_wizard => method(:get_wizard),
[C,pcsd/remote.rb,61]       :wizard_submit => method(:wizard_submit),
[C,pcsd/remote.rb,62]       :get_tokens => method(:get_tokens),
[C,pcsd/remote.rb,63]       :get_cluster_tokens => method(:get_cluster_tokens),
[C,pcsd/remote.rb,64]       :save_tokens => method(:save_tokens),
[C,pcsd/remote.rb,65]       :get_cluster_properties_definition => method(:get_cluster_properties_definition),
[C,pcsd/remote.rb,66]       :check_sbd => method(:check_sbd),
[C,pcsd/remote.rb,67]       :set_sbd_config => method(:set_sbd_config),
[C,pcsd/remote.rb,68]       :get_sbd_config => method(:get_sbd_config),
[C,pcsd/remote.rb,69]       :sbd_disable => method(:sbd_disable),
[C,pcsd/remote.rb,70]       :sbd_enable => method(:sbd_enable),
[C,pcsd/remote.rb,71]       :remove_stonith_watchdog_timeout=> method(:remove_stonith_watchdog_timeout),
[D,pcsd/remote.rb,72]       :set_stonith_watchdog_timeout_to_zero => method(:set_stonith_watchdog_timeout_to_zero)
[A,pcsd/remote.rb,72]       :set_stonith_watchdog_timeout_to_zero => method(:set_stonith_watchdog_timeout_to_zero),
[A,pcsd/remote.rb,73]       :remote_enable_sbd => method(:remote_enable_sbd),
[A,pcsd/remote.rb,74]       :remote_disable_sbd => method(:remote_disable_sbd)
[C,pcsd/remote.rb,75]   }
[C,pcsd/remote.rb,76]   remote_cmd_with_pacemaker = {
[C,pcsd/remote.rb,77]       :pacemaker_node_status => method(:remote_pacemaker_node_status),
[C,pcsd/remote.rb,78]       :resource_start => method(:resource_start),
[C,pcsd/remote.rb,79]       :resource_stop => method(:resource_stop),
[C,pcsd/remote.rb,80]       :resource_cleanup => method(:resource_cleanup),
[C,pcsd/remote.rb,81]       :update_resource => method(:update_resource),
[C,pcsd/remote.rb,82]       :update_fence_device => method(:update_fence_device),
[C,pcsd/remote.rb,83]       :get_avail_resource_agents => method(:get_avail_resource_agents),
[C,pcsd/remote.rb,84]       :get_avail_fence_agents => method(:get_avail_fence_agents),
[C,pcsd/remote.rb,85]       :remove_resource => method(:remove_resource),
[C,pcsd/remote.rb,86]       :add_constraint_remote => method(:add_constraint_remote),
[C,pcsd/remote.rb,87]       :add_constraint_rule_remote => method(:add_constraint_rule_remote),
[C,pcsd/remote.rb,88]       :add_constraint_set_remote => method(:add_constraint_set_remote),
[C,pcsd/remote.rb,89]       :remove_constraint_remote => method(:remove_constraint_remote),
@@ -764,31 +766,33 @@ end
[C,pcsd/remote.rb,766] 
[C,pcsd/remote.rb,767] def remote_add_node(params, request, auth_user, all=false)
[C,pcsd/remote.rb,768]   if not allowed_for_local_cluster(auth_user, Permissions::FULL)
[C,pcsd/remote.rb,769]     return 403, 'Permission denied'
[C,pcsd/remote.rb,770]   end
[C,pcsd/remote.rb,771]   auto_start = false
[C,pcsd/remote.rb,772]   if params[:auto_start] and params[:auto_start] == "1"
[C,pcsd/remote.rb,773]     auto_start = true
[C,pcsd/remote.rb,774]   end
[C,pcsd/remote.rb,775] 
[C,pcsd/remote.rb,776]   if params[:new_nodename] != nil
[C,pcsd/remote.rb,777]     node = params[:new_nodename]
[C,pcsd/remote.rb,778]     if params[:new_ring1addr] != nil
[C,pcsd/remote.rb,779]       node += ',' + params[:new_ring1addr]
[C,pcsd/remote.rb,780]     end
[D,pcsd/remote.rb,779]     retval, output = add_node(auth_user, node, all, auto_start)
[A,pcsd/remote.rb,781]     retval, output = add_node(
[A,pcsd/remote.rb,782]       auth_user, node, all, auto_start, params[:watchdog]
[A,pcsd/remote.rb,783]     )
[C,pcsd/remote.rb,784]   end
[C,pcsd/remote.rb,785] 
[C,pcsd/remote.rb,786]   if retval == 0
[C,pcsd/remote.rb,787]     return [200, JSON.generate([retval, get_corosync_conf()])]
[C,pcsd/remote.rb,788]   end
[C,pcsd/remote.rb,789] 
[C,pcsd/remote.rb,790]   return [400,output]
[C,pcsd/remote.rb,791] end
[C,pcsd/remote.rb,792] 
[C,pcsd/remote.rb,793] def remote_remove_nodes(params, request, auth_user)
[C,pcsd/remote.rb,794]   if not allowed_for_local_cluster(auth_user, Permissions::FULL)
[C,pcsd/remote.rb,795]     return 403, 'Permission denied'
[C,pcsd/remote.rb,796]   end
[C,pcsd/remote.rb,797]   count = 0
[C,pcsd/remote.rb,798]   out = ""
@@ -2287,15 +2291,71 @@ end
[C,pcsd/remote.rb,2291] 
[C,pcsd/remote.rb,2292] def set_stonith_watchdog_timeout_to_zero(param, request, auth_user)
[C,pcsd/remote.rb,2293]   unless allowed_for_local_cluster(auth_user, Permissions::WRITE)
[C,pcsd/remote.rb,2294]     return 403, 'Permission denied'
[C,pcsd/remote.rb,2295]   end
[C,pcsd/remote.rb,2296]   if set_cluster_prop_force(auth_user, 'stonith-watchdog-timeout', '0')
[C,pcsd/remote.rb,2297]     $logger.info('Cluster property "stonith-watchdog-timeout" set to "0"')
[C,pcsd/remote.rb,2298]     return [200, 'OK']
[C,pcsd/remote.rb,2299]   else
[C,pcsd/remote.rb,2300]     $logger.info(
[C,pcsd/remote.rb,2301]       'Failed to set cluster property "stonith-watchdog-timeout"to 0'
[C,pcsd/remote.rb,2302]     )
[C,pcsd/remote.rb,2303]     return [400, 'ERROR']
[C,pcsd/remote.rb,2304]   end
[C,pcsd/remote.rb,2305] end
[A,pcsd/remote.rb,2306] 
[A,pcsd/remote.rb,2307] def remote_enable_sbd(params, request, auth_user)
[A,pcsd/remote.rb,2308]   unless allowed_for_local_cluster(auth_user, Permissions::WRITE)
[A,pcsd/remote.rb,2309]     return 403, 'Permission denied'
[A,pcsd/remote.rb,2310]   end
[A,pcsd/remote.rb,2311] 
[A,pcsd/remote.rb,2312]   arg_list = []
[A,pcsd/remote.rb,2313] 
[A,pcsd/remote.rb,2314]   if ['true', '1', 'on'].include?(params[:ignore_offline_nodes])
[A,pcsd/remote.rb,2315]     arg_list << '--skip-offline'
[A,pcsd/remote.rb,2316]   end
[A,pcsd/remote.rb,2317] 
[A,pcsd/remote.rb,2318]   params[:watchdog].each do |node, watchdog|
[A,pcsd/remote.rb,2319]     unless watchdog.strip.empty?
[A,pcsd/remote.rb,2320]       arg_list << "--watchdog=#{watchdog.strip}@#{node}"
[A,pcsd/remote.rb,2321]     end
[A,pcsd/remote.rb,2322]   end
[A,pcsd/remote.rb,2323] 
[A,pcsd/remote.rb,2324]   params[:config].each do |option, value|
[A,pcsd/remote.rb,2325]     unless value.empty?
[A,pcsd/remote.rb,2326]       arg_list << "#{option}=#{value}"
[A,pcsd/remote.rb,2327]     end
[A,pcsd/remote.rb,2328]   end
[A,pcsd/remote.rb,2329] 
[A,pcsd/remote.rb,2330]   _, stderr, retcode = run_cmd(
[A,pcsd/remote.rb,2331]     auth_user, PCS, 'stonith', 'sbd', 'enable', *arg_list
[A,pcsd/remote.rb,2332]   )
[A,pcsd/remote.rb,2333] 
[A,pcsd/remote.rb,2334]   if retcode != 0
[A,pcsd/remote.rb,2335]     return [400, "Unable to enable sbd in cluster: #{stderr.join('')}"]
[A,pcsd/remote.rb,2336]   end
[A,pcsd/remote.rb,2337] 
[A,pcsd/remote.rb,2338]   return [200, 'Sbd has been enabled.']
[A,pcsd/remote.rb,2339] end
[A,pcsd/remote.rb,2340] 
[A,pcsd/remote.rb,2341] def remote_disable_sbd(params, request, auth_user)
[A,pcsd/remote.rb,2342]   unless allowed_for_local_cluster(auth_user, Permissions::WRITE)
[A,pcsd/remote.rb,2343]     return 403, 'Permission denied'
[A,pcsd/remote.rb,2344]   end
[A,pcsd/remote.rb,2345] 
[A,pcsd/remote.rb,2346]   arg_list = []
[A,pcsd/remote.rb,2347] 
[A,pcsd/remote.rb,2348]   if ['true', '1', 'on'].include?(params[:ignore_offline_nodes])
[A,pcsd/remote.rb,2349]     arg_list << '--skip-offline'
[A,pcsd/remote.rb,2350]   end
[A,pcsd/remote.rb,2351] 
[A,pcsd/remote.rb,2352]   _, stderr, retcode = run_cmd(
[A,pcsd/remote.rb,2353]     auth_user, PCS, 'stonith', 'sbd', 'disable', *arg_list
[A,pcsd/remote.rb,2354]   )
[A,pcsd/remote.rb,2355] 
[A,pcsd/remote.rb,2356]   if retcode != 0
[A,pcsd/remote.rb,2357]     return [400, "Unable to disable sbd in cluster: #{stderr.join('')}"]
[A,pcsd/remote.rb,2358]   end
[A,pcsd/remote.rb,2359] 
[A,pcsd/remote.rb,2360]   return [200, 'Sbd has been disabled.']
[A,pcsd/remote.rb,2361] end
@@ -43,15 +43,175 @@
[C,pcsd/views/_dialogs.erb,43]     <p style="color: orange">This is a CMAN cluster with UDPU transport, cluster restart is required to apply node removal.</p>
[C,pcsd/views/_dialogs.erb,44]   {{/if}}
[C,pcsd/views/_dialogs.erb,45] </div>
[C,pcsd/views/_dialogs.erb,46] 
[C,pcsd/views/_dialogs.erb,47] <div id="dialog_verify_remove_resources" style="display: none;">
[C,pcsd/views/_dialogs.erb,48]   <p style="font-size:12px;">Are you sure you want to remove the following resource(s)?</p>
[C,pcsd/views/_dialogs.erb,49]   <span class="name_list"></span>
[C,pcsd/views/_dialogs.erb,50]   <input type="checkbox" name="force">
[C,pcsd/views/_dialogs.erb,51]   Enforce removal (Remove the resources without stopping them first.)
[C,pcsd/views/_dialogs.erb,52] </div>
[C,pcsd/views/_dialogs.erb,53] 
[C,pcsd/views/_dialogs.erb,54] <div id="dialog_verify_remove_acl_roles" style="display:none;">
[C,pcsd/views/_dialogs.erb,55]   <p style="font-size:12px;">Are you sure you want to remove the following ACL role(s)?</p>
[C,pcsd/views/_dialogs.erb,56]   <span class="name_list"></span>
[C,pcsd/views/_dialogs.erb,57] </div>
[A,pcsd/views/_dialogs.erb,58] 
[A,pcsd/views/_dialogs.erb,59] <div id="enable_sbd_dialog" style="display:none;">
[A,pcsd/views/_dialogs.erb,60]   <form id="enable_sbd_form">
[A,pcsd/views/_dialogs.erb,61]     <table>
[A,pcsd/views/_dialogs.erb,62]       <tr>
[A,pcsd/views/_dialogs.erb,63]         <td class="darkdatatable_header">
[A,pcsd/views/_dialogs.erb,64]           Specify watchdog devices for nodes:
[A,pcsd/views/_dialogs.erb,65]         </td>
[A,pcsd/views/_dialogs.erb,66]       </tr>
[A,pcsd/views/_dialogs.erb,67]       <tr>
[A,pcsd/views/_dialogs.erb,68]         <td>
[A,pcsd/views/_dialogs.erb,69]           <table id="watchdog_table">
[A,pcsd/views/_dialogs.erb,70]           </table>
[A,pcsd/views/_dialogs.erb,71]         </td>
[A,pcsd/views/_dialogs.erb,72]       </tr>
[A,pcsd/views/_dialogs.erb,73]     </table>
[A,pcsd/views/_dialogs.erb,74]     <table style="margin-top: 20px;">
[A,pcsd/views/_dialogs.erb,75]       <tr>
[A,pcsd/views/_dialogs.erb,76]         <td class="darkdatatable_header">
[A,pcsd/views/_dialogs.erb,77]           SBD options:
[A,pcsd/views/_dialogs.erb,78]         </td>
[A,pcsd/views/_dialogs.erb,79]       </tr>
[A,pcsd/views/_dialogs.erb,80]       <tr>
[A,pcsd/views/_dialogs.erb,81]         <td>
[A,pcsd/views/_dialogs.erb,82]           <table id="sbd_options">
[A,pcsd/views/_dialogs.erb,83]             <tr>
[A,pcsd/views/_dialogs.erb,84]               <td>SBD_DELAY_START</td>
[A,pcsd/views/_dialogs.erb,85]               <td>
[A,pcsd/views/_dialogs.erb,86]                 <select name="config[SBD_DELAY_START]">
[A,pcsd/views/_dialogs.erb,87]                   <option value="" selected>(default)</option>
[A,pcsd/views/_dialogs.erb,88]                   <option value="yes">yes</option>
[A,pcsd/views/_dialogs.erb,89]                   <option value="no">no</option>
[A,pcsd/views/_dialogs.erb,90]                 </select>
[A,pcsd/views/_dialogs.erb,91]               </td>
[A,pcsd/views/_dialogs.erb,92]             </tr>
[A,pcsd/views/_dialogs.erb,93]             <tr>
[A,pcsd/views/_dialogs.erb,94]               <td>SBD_STARTMODE</td>
[A,pcsd/views/_dialogs.erb,95]               <td>
[A,pcsd/views/_dialogs.erb,96]                 <select name="config[SBD_STARTMODE]">
[A,pcsd/views/_dialogs.erb,97]                   <option value="" selected>(default)</option>
[A,pcsd/views/_dialogs.erb,98]                   <option value="clean">clean</option>
[A,pcsd/views/_dialogs.erb,99]                   <option value="always">always</option>
[A,pcsd/views/_dialogs.erb,100]                 </select>
[A,pcsd/views/_dialogs.erb,101]               </td>
[A,pcsd/views/_dialogs.erb,102]             </tr>
[A,pcsd/views/_dialogs.erb,103]             <tr>
[A,pcsd/views/_dialogs.erb,104]               <td>SBD_WATCHDOG_TIMEOUT</td>
[A,pcsd/views/_dialogs.erb,105]               <td>
[A,pcsd/views/_dialogs.erb,106]                 <input
[A,pcsd/views/_dialogs.erb,107]                     type="text"
[A,pcsd/views/_dialogs.erb,108]                     name="config[SBD_WATCHDOG_TIMEOUT]"
[A,pcsd/views/_dialogs.erb,109]                     placeholder="5"
[A,pcsd/views/_dialogs.erb,110]                 />
[A,pcsd/views/_dialogs.erb,111]               </td>
[A,pcsd/views/_dialogs.erb,112]             </tr>
[A,pcsd/views/_dialogs.erb,113]           </table>
[A,pcsd/views/_dialogs.erb,114]         </td>
[A,pcsd/views/_dialogs.erb,115]       </tr>
[A,pcsd/views/_dialogs.erb,116]     </table>
[A,pcsd/views/_dialogs.erb,117]     <table>
[A,pcsd/views/_dialogs.erb,118]       <tr>
[A,pcsd/views/_dialogs.erb,119]         <td>
[A,pcsd/views/_dialogs.erb,120]           <label for="ignore_offline_nodes_enable_sbd">
[A,pcsd/views/_dialogs.erb,121]             ignore offline nodes
[A,pcsd/views/_dialogs.erb,122]           </label>
[A,pcsd/views/_dialogs.erb,123]         </td>
[A,pcsd/views/_dialogs.erb,124]         <td>
[A,pcsd/views/_dialogs.erb,125]           <input
[A,pcsd/views/_dialogs.erb,126]               id="ignore_offline_nodes_enable_sbd"
[A,pcsd/views/_dialogs.erb,127]               type="checkbox"
[A,pcsd/views/_dialogs.erb,128]               name="ignore_offline_nodes"
[A,pcsd/views/_dialogs.erb,129]           />
[A,pcsd/views/_dialogs.erb,130]         </td>
[A,pcsd/views/_dialogs.erb,131]       </tr>
[A,pcsd/views/_dialogs.erb,132]     </table>
[A,pcsd/views/_dialogs.erb,133]   </form>
[A,pcsd/views/_dialogs.erb,134] </div>
[A,pcsd/views/_dialogs.erb,135] 
[A,pcsd/views/_dialogs.erb,136] <div id="disable_sbd_dialog" style="display:none;">
[A,pcsd/views/_dialogs.erb,137]   <form id="disable_sbd_form">
[A,pcsd/views/_dialogs.erb,138]     Are you sure you want to disable SBD?
[A,pcsd/views/_dialogs.erb,139]     <table>
[A,pcsd/views/_dialogs.erb,140]       <tr>
[A,pcsd/views/_dialogs.erb,141]         <td>
[A,pcsd/views/_dialogs.erb,142]           <label for="ignore_offline_nodes_disable_sbd">
[A,pcsd/views/_dialogs.erb,143]             ignore offline nodes
[A,pcsd/views/_dialogs.erb,144]           </label>
[A,pcsd/views/_dialogs.erb,145]         </td>
[A,pcsd/views/_dialogs.erb,146]         <td>
[A,pcsd/views/_dialogs.erb,147]           <input
[A,pcsd/views/_dialogs.erb,148]               id="ignore_offline_nodes_disable_sbd"
[A,pcsd/views/_dialogs.erb,149]               type="checkbox"
[A,pcsd/views/_dialogs.erb,150]               name="ignore_offline_nodes"
[A,pcsd/views/_dialogs.erb,151]           />
[A,pcsd/views/_dialogs.erb,152]         </td>
[A,pcsd/views/_dialogs.erb,153]       </tr>
[A,pcsd/views/_dialogs.erb,154]     </table>
[A,pcsd/views/_dialogs.erb,155]   </form>
[A,pcsd/views/_dialogs.erb,156] </div>
[A,pcsd/views/_dialogs.erb,157] 
[A,pcsd/views/_dialogs.erb,158] <div id="sbd_status_dialog" style="display:none;">
[A,pcsd/views/_dialogs.erb,159]   <table>
[A,pcsd/views/_dialogs.erb,160]     <tr>
[A,pcsd/views/_dialogs.erb,161]       <td class="darkdatatable_header">SBD service status</td>
[A,pcsd/views/_dialogs.erb,162]     </tr>
[A,pcsd/views/_dialogs.erb,163]     <tr>
[A,pcsd/views/_dialogs.erb,164]       <td>
[A,pcsd/views/_dialogs.erb,165]         <table class="darkdatatable">
[A,pcsd/views/_dialogs.erb,166]           <tr>
[A,pcsd/views/_dialogs.erb,167]             <th>NODE</th>
[A,pcsd/views/_dialogs.erb,168]             <th>INSTALLED</th>
[A,pcsd/views/_dialogs.erb,169]             <th>ENABLED</th>
[A,pcsd/views/_dialogs.erb,170]             <th>RUNNING</th>
[A,pcsd/views/_dialogs.erb,171]           </tr>
[A,pcsd/views/_dialogs.erb,172]           {{#each node in Pcs.nodesController}}
[A,pcsd/views/_dialogs.erb,173]           <tr>
[A,pcsd/views/_dialogs.erb,174]             <td>{{node.name}}</td>
[A,pcsd/views/_dialogs.erb,175]             <td>{{bool-to-icon node.is_sbd_installed}}</td>
[A,pcsd/views/_dialogs.erb,176]             <td>{{bool-to-icon node.is_sbd_enabled}}</td>
[A,pcsd/views/_dialogs.erb,177]             <td>{{bool-to-icon node.is_sbd_running}}</td>
[A,pcsd/views/_dialogs.erb,178]           </tr>
[A,pcsd/views/_dialogs.erb,179]           {{/each}}
[A,pcsd/views/_dialogs.erb,180]         </table>
[A,pcsd/views/_dialogs.erb,181]       </td>
[A,pcsd/views/_dialogs.erb,182]     </tr>
[A,pcsd/views/_dialogs.erb,183]   </table>
[A,pcsd/views/_dialogs.erb,184]   {{#if Pcs.is_sbd_enabled_or_running}}
[A,pcsd/views/_dialogs.erb,185]   <table style="margin-top: 20px;">
[A,pcsd/views/_dialogs.erb,186]     <tr>
[A,pcsd/views/_dialogs.erb,187]       <td class="darkdatatable_header">SBD configuration</td>
[A,pcsd/views/_dialogs.erb,188]     </tr>
[A,pcsd/views/_dialogs.erb,189]     <tr>
[A,pcsd/views/_dialogs.erb,190]       <td>
[A,pcsd/views/_dialogs.erb,191]         {{{Pcs.sbd_config_table}}}
[A,pcsd/views/_dialogs.erb,192]       </td>
[A,pcsd/views/_dialogs.erb,193]     </tr>
[A,pcsd/views/_dialogs.erb,194]   </table>
[A,pcsd/views/_dialogs.erb,195]   <table style="margin-top: 20px;">
[A,pcsd/views/_dialogs.erb,196]     <tr>
[A,pcsd/views/_dialogs.erb,197]       <td class="darkdatatable_header">SBD watchdogs</td>
[A,pcsd/views/_dialogs.erb,198]     </tr>
[A,pcsd/views/_dialogs.erb,199]     <tr>
[A,pcsd/views/_dialogs.erb,200]       <td>
[A,pcsd/views/_dialogs.erb,201]         <table class="darkdatatable">
[A,pcsd/views/_dialogs.erb,202]           <tr>
[A,pcsd/views/_dialogs.erb,203]             <th>NODE</th>
[A,pcsd/views/_dialogs.erb,204]             <th>WATCHDOG</th>
[A,pcsd/views/_dialogs.erb,205]           </tr>
[A,pcsd/views/_dialogs.erb,206]           {{#each node in Pcs.nodesController}}
[A,pcsd/views/_dialogs.erb,207]           <tr>
[A,pcsd/views/_dialogs.erb,208]             <td>{{node.name}}</td>
[A,pcsd/views/_dialogs.erb,209]             <td>{{node.sbd_watchdog}}</td>
[A,pcsd/views/_dialogs.erb,210]           </tr>
[A,pcsd/views/_dialogs.erb,211]           {{/each}}
[A,pcsd/views/_dialogs.erb,212]         </table>
[A,pcsd/views/_dialogs.erb,213]       </td>
[A,pcsd/views/_dialogs.erb,214]     </tr>
[A,pcsd/views/_dialogs.erb,215]   </table>
[A,pcsd/views/_dialogs.erb,216]   {{/if}}
[A,pcsd/views/_dialogs.erb,217] </div>
@@ -9,30 +9,38 @@
[C,pcsd/views/_resource.erb,9] 	<% if @myView == "resource" %>
[C,pcsd/views/_resource.erb,10] 	  <a href="#" onclick='verify_remove_resources(); return false;'>
[C,pcsd/views/_resource.erb,11] 	<% else %>
[C,pcsd/views/_resource.erb,12] 	  <a href="#" onclick='verify_remove_fence_devices(); return false;'>
[C,pcsd/views/_resource.erb,13] 	<% end %>
[C,pcsd/views/_resource.erb,14]       Remove</a>&nbsp;&nbsp;&nbsp;&nbsp;</div>
[C,pcsd/views/_resource.erb,15]   <div class="plus sprites"></div><div class="link">&nbsp;
[C,pcsd/views/_resource.erb,16]     <% if @myView == "resource" %>
[C,pcsd/views/_resource.erb,17]       <a href="#" onclick="$('#add_resource').dialog({title: 'Add Resource', modal:true, width: 'auto'});return false;">
[C,pcsd/views/_resource.erb,18]     <% else %>
[C,pcsd/views/_resource.erb,19]       <a href="#" onclick="$('#add_stonith').dialog({title: 'Add Fence Device', modal:true, width: 'auto'});return false;">
[C,pcsd/views/_resource.erb,20]     <% end %>
[C,pcsd/views/_resource.erb,21]   Add</a>&nbsp;&nbsp;&nbsp;&nbsp;</div>
[C,pcsd/views/_resource.erb,22]     <% if @myView == "resource" %>
[C,pcsd/views/_resource.erb,23]       <div class="plus sprites"></div><div class="link">&nbsp;<a href="#" onclick="create_group();return false;">Create Group</a></div>
[A,pcsd/views/_resource.erb,24]     <% else %>
[A,pcsd/views/_resource.erb,25]         {{#if Pcs.is_sbd_supported}}
[A,pcsd/views/_resource.erb,26]         <div class="configurelight sprites"></div>
[A,pcsd/views/_resource.erb,27]         <div class="link">
[A,pcsd/views/_resource.erb,28]           &nbsp;
[A,pcsd/views/_resource.erb,29]           <a href="#" onclick="sbd_status_dialog();return false;">SBD</a>
[A,pcsd/views/_resource.erb,30]         </div>
[A,pcsd/views/_resource.erb,31]         {{/if}}
[C,pcsd/views/_resource.erb,32]     <% end %>
[C,pcsd/views/_resource.erb,33]     </td>
[C,pcsd/views/_resource.erb,34]     <td class="borderbottom">&nbsp;</td>
[C,pcsd/views/_resource.erb,35]     <td id="tab_name" width="125">
[C,pcsd/views/_resource.erb,36]       <div style="float: left">&nbsp;Edit <%= @pageName%>&nbsp;</div><div class="check sprites"></div>
[C,pcsd/views/_resource.erb,37]     </td>
[C,pcsd/views/_resource.erb,38]   </tr>
[C,pcsd/views/_resource.erb,39]   <tr id="<%=@myView%>_list_row" {{bind-attr style="Pcs.<%=@myView%>_page"}}>
[C,pcsd/views/_resource.erb,40]     <td id="<%=@myView%>_list" class="node_list">
[C,pcsd/views/_resource.erb,41]       <%= erb :_resource_list %>
[C,pcsd/views/_resource.erb,42]     </td>
[C,pcsd/views/_resource.erb,43]   <td id="node_info" colspan=2>
[C,pcsd/views/_resource.erb,44]     <div id="<%=@myView%>_info_div">
[C,pcsd/views/_resource.erb,45]     <% if @myView == "resource" %>
[C,pcsd/views/_resource.erb,46]       {{resource-edit resource=Pcs.resourcesContainer.cur_resource page_name="Resource" old_pcsd=Pcs.resourcesContainer.is_version_1 utilization_support=Pcs.nodesController.utilization_support}}
@@ -173,30 +173,35 @@
[C,pcsd/views/nodes.erb,173]                       <span id="pacemaker_status" style="float:left" class="status-unknown">Unknown ({{Pcs.nodesController.cur_node.pacemaker_startup}})</span>
[C,pcsd/views/nodes.erb,174]                       {{/if}}
[C,pcsd/views/nodes.erb,175]                       {{/if}}
[C,pcsd/views/nodes.erb,176]                     </td></tr>
[C,pcsd/views/nodes.erb,177]                     <tr><td>corosync</td><td><div style="float:left;margin-right:6px;" class="check sprites"></div>
[C,pcsd/views/nodes.erb,178]                       {{#if Pcs.nodesController.cur_node.corosync_daemon}}
[C,pcsd/views/nodes.erb,179]                       <span id="corosync_status" style="float:left" class="status">Running ({{Pcs.nodesController.cur_node.corosync_startup}})</span>
[C,pcsd/views/nodes.erb,180]                       {{else}}
[C,pcsd/views/nodes.erb,181]                       {{#if Pcs.nodesController.cur_node.pcsd}}
[C,pcsd/views/nodes.erb,182]                       <span id="corosync_status" style="float:left" class="status-offline">Stopped ({{Pcs.nodesController.cur_node.corosync_startup}})</span>
[C,pcsd/views/nodes.erb,183]                       {{else}}
[C,pcsd/views/nodes.erb,184]                       <span id="corosync_status" style="float:left" class="status-unknown">Unknown ({{Pcs.nodesController.cur_node.corosync_startup}})</span>
[C,pcsd/views/nodes.erb,185]                       {{/if}}
[C,pcsd/views/nodes.erb,186]                       {{/if}}
[C,pcsd/views/nodes.erb,187]                     </td></tr>
[A,pcsd/views/nodes.erb,188]                     {{#if Pcs.is_sbd_enabled_or_running }}
[A,pcsd/views/nodes.erb,189]                     <tr><td>sbd</td><td><div style="float:left;margin-right:6px;" class="check sprites"></div>
[A,pcsd/views/nodes.erb,190]                       {{{Pcs.nodesController.cur_node.sbd_status_str}}}
[A,pcsd/views/nodes.erb,191]                     </td></tr>
[A,pcsd/views/nodes.erb,192]                     {{/if}}
[C,pcsd/views/nodes.erb,193]                     <tr><td>pcsd</td><td><div style="float:left;margin-right:6px;" class="check sprites"></div>
[C,pcsd/views/nodes.erb,194]                       {{#if Pcs.nodesController.cur_node.pcsd}}
[C,pcsd/views/nodes.erb,195]                       <span id="pcsd_status" style="float:left" class="status">Running ({{Pcs.nodesController.cur_node.pcsd_startup}})</span>
[C,pcsd/views/nodes.erb,196]                       {{else}}
[C,pcsd/views/nodes.erb,197]                       {{#if Pcs.nodesController.cur_node.authorized}}
[C,pcsd/views/nodes.erb,198]                       <span id="pcsd_status" style="float:left" class="status-offline">Stopped ({{Pcs.nodesController.cur_node.pcsd_startup}})</span>
[C,pcsd/views/nodes.erb,199]                       {{else}}
[C,pcsd/views/nodes.erb,200]                       <span id="pcsd_status" style="float:left" class="status-offline">Running (not Authorized) ({{Pcs.nodesController.cur_node.pcsd_startup}})</span>
[C,pcsd/views/nodes.erb,201]                       {{/if}}
[C,pcsd/views/nodes.erb,202]                       {{/if}}
[C,pcsd/views/nodes.erb,203]                     </td></tr>
[C,pcsd/views/nodes.erb,204]                   </table>
[C,pcsd/views/nodes.erb,205]                 </div>
[C,pcsd/views/nodes.erb,206]               </td>
[C,pcsd/views/nodes.erb,207]               </tr>
@@ -323,23 +328,31 @@
[C,pcsd/views/nodes.erb,328]     <table id="add_node_selector">
[C,pcsd/views/nodes.erb,329]       <tr>
[C,pcsd/views/nodes.erb,330]         <td>Node Name:</td>
[C,pcsd/views/nodes.erb,331]         <td><input type=text name=new_nodename></td>
[C,pcsd/views/nodes.erb,332]       </tr>
[C,pcsd/views/nodes.erb,333]       {{#if Pcs.need_ring1_address}}
[C,pcsd/views/nodes.erb,334]       <tr>
[C,pcsd/views/nodes.erb,335]         <td>Ring1 address:</td>
[C,pcsd/views/nodes.erb,336]         <td><input type=text name=new_ring1addr></td>
[C,pcsd/views/nodes.erb,337]       </tr>
[C,pcsd/views/nodes.erb,338]       {{/if}}
[C,pcsd/views/nodes.erb,339]       <tr>
[C,pcsd/views/nodes.erb,340]         <td>Auto Start/Enable:</td>
[C,pcsd/views/nodes.erb,341]         <td><input type="checkbox" name="auto_start" value="1" checked></td>
[C,pcsd/views/nodes.erb,342]       </tr>
[A,pcsd/views/nodes.erb,343]       {{#if Pcs.is_sbd_enabled}}
[A,pcsd/views/nodes.erb,344]       <tr>
[A,pcsd/views/nodes.erb,345]         <td>SBD watchdog:</td>
[A,pcsd/views/nodes.erb,346]         <td>
[A,pcsd/views/nodes.erb,347]           <input type="text" name="watchdog" placeholder="/dev/watchdog" />
[A,pcsd/views/nodes.erb,348]         </td>
[A,pcsd/views/nodes.erb,349]       </tr>
[A,pcsd/views/nodes.erb,350]       {{/if}}
[C,pcsd/views/nodes.erb,351]       {{#if Pcs.is_cman_with_udpu_transport}}
[C,pcsd/views/nodes.erb,352]       <tr>
[C,pcsd/views/nodes.erb,353]         <td colspan="2" style="color: orange">This is a CMAN cluster with UDPU transport,<br>cluster restart is required to apply node addition.</td>
[C,pcsd/views/nodes.erb,354]       </tr>
[C,pcsd/views/nodes.erb,355]       {{/if}}
[C,pcsd/views/nodes.erb,356]     </table>
[C,pcsd/views/nodes.erb,357]   </form>
[C,pcsd/views/nodes.erb,358] </div>